### ISSUE TITLE
wss:// requests forcing interceptor of selenium-wire into a infinite loop

### ISSUE THREAD

**Issue by Avnsx** [wkeeling/selenium-wire#540]
I am trying to Intercept a specific request on [a website](https://fansly.com/). Once you've created a account on it and are browsing with that account; the website seems to launch a wss request (most likely for the onsite chat feature). This wss request is putting selenium-wire into a infinite loop, either constantly triggering it or triggering the couple requests that would come before or after it (but all those requests erroring out). This in return never allows the website to launch all requests in a normally successful order. To be honest, I am not too sure if it's the wss request alone or the seemingly failing ssl certificate verification. I already tried doing ``verify=False`` in selenium-wire options that didn't change anything. Loading the website without my own user data dir / without a account on the website, the wss request seems to pass through, most likely because you can't connect with the websocket request if you are not authenticated (have a user account on the website).

Here's a visual representation of said issue: https://youtu.be/glnykmuG-cw

Please help me out 🙏 @wkeeling 

```python
import seleniumwire.undetected_chromedriver as uc
import re, os

chrome_options = uc.ChromeOptions()

# after creating a account on the website I am copying local data from browser here;
# so that I can browse the website with a authenticated selenium session
for e in ['--disable-extensions', fr'--user-data-dir={os.getenv("localappdata")}\Google\Chrome\User Data']:
    chrome_options.add_argument(e)
chrome_options.headless = False

driver = uc.Chrome(options=chrome_options, use_subprocess=True)

def interceptor(request):
    print(request.url)

driver.request_interceptor = interceptor
driver.get('https://fansly.com/')
input()
```

**Selenium-wire version 4.6.3 | UC version 3.1.5.post4 | Python version 3.10.4 | Windows 10**

**Comment by randolfvalerija** [comment]
@wkeeling I have the same issue here. The site cannot load because the websocket messages are not intercepted correctly? If I disable request capture altogether, the site loads (well, the ws messages). 

- url: https://bexar.tx.publicsearch.us/results?department=RP&docTypes=DEED&recordedDateRange=20220303%2C20220303&searchType=advancedSearch
- swire-version: 4.6.3
- selenium-version: 4.1.3
- chrome: 100 (but also tested on the, as of writing, latest firefox)

logs:

> [2022-05-10 19:25:04] INFO:seleniumwire.handler:Capturing request: wss://bexar.tx.publicsearch.us/ws
> [2022-05-10 19:25:04] DEBUG:seleniumwire.server:127.0.0.1:49772: serverdisconnect
>   -> ('104.144.125.137', 4444)
> [2022-05-10 19:25:04] DEBUG:seleniumwire.server:127.0.0.1:49772: serverconnect
>   -> ('104.144.125.137', 4444)
> [2022-05-10 19:25:04] DEBUG:seleniumwire.server:127.0.0.1:49772: Sending CONNECT request
>   -> Proxy Server: ('104.144.125.137', 4444)
>   -> Connect to: bexar.tx.publicsearch.us:80
> [2022-05-10 19:25:04] DEBUG:seleniumwire.server:127.0.0.1:49772: response
>   -> Response(400, text/html, 654b)
> [2022-05-10 19:25:04] INFO:seleniumwire.handler:Capturing response: wss://bexar.tx.publicsearch.us:80/ws 400 Bad Request
> [2022-05-10 19:25:04] DEBUG:seleniumwire.server:127.0.0.1:49772: serverdisconnect
>   -> ('104.144.125.137', 4444)
> [2022-05-10 19:25:04] DEBUG:seleniumwire.server:127.0.0.1:49772: clientdisconnect
> [2022-05-10 19:25:07] DEBUG:seleniumwire.server:127.0.0.1:49774: clientconnect
> [2022-05-10 19:25:07] DEBUG:seleniumwire.server:127.0.0.1:49774: serverconnect
>   -> ('104.144.125.137', 4444)
> [2022-05-10 19:25:07] DEBUG:seleniumwire.server:127.0.0.1:49774: Establish TLS with server
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: ALPN selected by server: http/1.1
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: Establish TLS with client
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: ALPN for client: b'http/1.1'
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: request
>   -> Request(GET /ws)
> thi is the request_interceptor!!
> wss://bexar.tx.publicsearch.us/ws
> [2022-05-10 19:25:08] INFO:seleniumwire.handler:Capturing request: wss://bexar.tx.publicsearch.us/ws
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: serverdisconnect
>   -> ('104.144.125.137', 4444)
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: serverconnect
>   -> ('104.144.125.137', 4444)
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: Sending CONNECT request
>   -> Proxy Server: ('104.144.125.137', 4444)
>   -> Connect to: bexar.tx.publicsearch.us:80
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: response
>   -> Response(400, text/html, 654b)
> [2022-05-10 19:25:08] INFO:seleniumwire.handler:Capturing response: wss://bexar.tx.publicsearch.us:80/ws 400 Bad Request
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: serverdisconnect
>   -> ('104.144.125.137', 4444)
> [2022-05-10 19:25:08] DEBUG:seleniumwire.server:127.0.0.1:49774: clientdisconnect
> [2022-05-10 19:25:11] DEBUG:seleniumwire.server:127.0.0.1:49776: clientconnect
> [2022-05-10 19:25:11] DEBUG:seleniumwire.server:127.0.0.1:49776: serverconnect
>   -> ('104.144.125.137', 4444)

**Comment by wkeeling** [comment]
@randolfvalerija 

> I was hoping to use the request interceptor to catch the ws messages on the fly.

Yes this isn't currently possible for websocket messages. You'll need to wait for the request/response to fully complete, then access the messages via the `ws_messages` attribute.

**Comment by wkeeling** [comment]
A fix is now available in version 4.6.4.

**Comment by Avnsx** [comment]
> A fix is now available in version 4.6.4.

Hey thanks for the quick response! But this does not seem to entirely fix my issue. The website I mentioned seems to load another wss request after the first one and it actually still gets stuck there. Also apart from that all requests are failing saying ``(failed) net::ERR_CERT_AUTHORITY_INVALID`` or ``net::ERR_FAILED``. This only happens on this specific website for me and only when I have registered a account on the website before and set selenium-wire to use my browser data and browse that website with it in a authenticated session. How could I fix this?

**Comment by wkeeling** [comment]
@Avnsx sorry for the delay coming back on this. I've tried reproducing by creating a fansly.com account, signing in manually and then pointing Selenium Wire/undetected chromedriver at the user data folder. Chrome opens fansly.com up straight into the logged in session as expected. Checking the network panel and all looks well - there seems to be no errors relating to any requests as far as I can see.

The only difference is that I'm running on Linux and it looks as though you're using Windows. The user data folder on Linux lives in `~/.config/google-chrome/`

My code is pretty much the same as yours:
```python
chrome_options = uc.ChromeOptions()
for e in [
    '--disable-extensions',
    '--user-data-dir=/tmp/google-chrome',  # Contains the logged in session
    '--profile-directory=Profile_1'
]:
    chrome_options.add_argument(e)

driver = uc.Chrome(options=chrome_options, use_subprocess=True)
driver.get('https://fansly.com/')
input()

driver.quit()
```
